backend/pages/utils/data_processing.py

```
@@ -3,23 +3,27 @@
     df_clean = df_clean.replace('X', 1)
 
-    for col in df_clean.columns:
-        df_clean[col] = pd.to_numeric(df_clean[col], errors='ignore')
+    # Replace NaN with None (which becomes `null` in JSON)
+    df_clean = df_clean.replace({np.nan: None})
 
-    # Round the floats to 5 digits 
+    try:
+        for col in df_clean.columns:
+            df_clean[col] = pd.to_numeric(df_clean[col], errors='ignore')
+    except:
```

Ruff (0.11.9)

Do not use bare `except`

(E722)